src/main/java/org/myrobotlab/service/AzureTranslator.java

```diff
@@ -169,14 +168,14 @@ public String translate(String toTranslate, String from, String to) {
       Response response = client.newCall(request).execute();
       String resp = response.body().string();
       // if ()
-      List<LinkedTreeMap> list = CodecUtils.fromJson(resp, List.class);
-      for (LinkedTreeMap t : list) {
-        LinkedTreeMap detected = (LinkedTreeMap) t.get("detectedLanguage");
+      List<LinkedHashMap> list = CodecUtils.fromJson(resp, List.class);

```

Comment:
why the change to LinkedHashMap ?

Reply:
default map class implemented by AP ...

Reply:
updated, honestly I don't know how the "left side" became non-generic
At some point in the past I know that you couldn't supply fromJson(json, XXX) with an interface.
I'm "guessing" this changed, otherwise it will blow up with List.class

Reply:
I think I know what the problem you're describing is, if you supply just an interface with no default implementation (as set when configuring the deserializers), GSON and Jackson don't know what to do since you can't construct an interface. But for List and Map they have default implementations since those are JSON "primitives". If we tried passing in an arbitrary interface, like `Serializable`, and disabled the polymorphic extensions I wrote the deserializers would blow up